code/game/objects/items/XMAS.dm

```diff
@@ -97,7 +97,6 @@
 	return
 
 /obj/item/weapon/gun/launcher/rocket/m57a4/XMAS
-	..()
 	flags_gun_features = GUN_INTERNAL_MAG
 	able_to_fire(mob/living/user)
```

Comment:
You should turn this into a full typepath proc like the contribution guidelines mention since you're cleaning up the code.